src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] array = new Object[10];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            array = copyArray(array);
```

Comment:
```suggestion
            resize();
```
ми можемо всередині методу переприсвоїти посилання